src/Compilers/CSharp/Portable/Compiler/DocumentationCommentCompiler.cs

```diff
@@ -276,7 +293,10 @@ public override void DefaultVisit(Symbol symbol)
             // If there are no doc comments, then no further work is required (other than to report a diagnostic if one is required).
             if (docCommentNodes.IsEmpty)
             {
-                if (maxDocumentationMode >= DocumentationMode.Diagnose && RequiresDocumentationComment(symbol))
+                if (maxDocumentationMode >= DocumentationMode.Diagnose
+                    && RequiresDocumentationComment(symbol)
+                    && !symbol.IsPartialImplementation()
+                    && !shouldSkipPartialDefinitionComments)
```

Comment:
Does this line matter? It looks like `!symbol.IsPartialImplementation()` implies `!shouldSkipPartialDefinitionComments`

Reply:
I think `symbol.IsPartialImplementation()` implies `!shouldSkipPartialDefinitionComments`.

In absence of the `!shouldSkipPartialDefinitionComments`, we give a warning in `ExtendedPartialMethods_MultipleFiles_ImplementationComment`.

In absence of the `!symbol.IsPartialImplementation()`, we give a warning in `ExtendedPartialMethods_MultipleFiles_DefinitionComment`, and give an extra warning (which is perhaps harmless) in `ExtendedPartialMethods_MultipleFiles_NoComment`.

The rule here is to never warn about absence of a doc comment on a partial method implementation, and to suppress a missing doc comment warning on a partial definition if we are not intending to output it anyway. Maybe I should add a comment.